web/src/lib/components/photos-page/asset-grid.svelte

```diff
@@ -71,7 +71,17 @@
   };
 
   const onDelete = () => {
-    if (!isTrashEnabled && $showDeleteModal) {
+    const hasTrashedAsset = Array.from($selectedAssets).some((asset) => asset.isTrashed);
+
+    if (hasTrashedAsset) {
+      if ($showDeleteModal) {
+        isShowDeleteConfirmation = true;
+        return;
+      } else {
+        handlePromiseError(trashOrDelete(true));
+        return;
+      }
+    } else if (!isTrashEnabled && $showDeleteModal) {
```

Comment:
When you're in the else case `isTrashEnabled` is already false, so the first part of the and is tautological.

Reply:
Why is it already false ?

Reply:
Oh nvm, I misread `hasTrashedAsset` in l. 76 for `isTrashEnabled`. Sorry!

Reply:
But your new commit still looks better haha

Reply:
> Oh nvm, I misread `hasTrashedAsset` in l. 76 for `isTrashEnabled`. Sorry!

np

> But your new commit still looks better haha

yeah, my first commit was a quick and ugly fix :sweat_smile: